csrc/runtime/allocations.cpp

```diff
@@ -741,13 +757,22 @@ at::Tensor transformFromAllocationToLogical(
                .run(logical, alloc);
   NVF_ERROR(frontier.size() == logical.size());
 
-  // give up on producing right shape/stride when allocation domain has
+  // give up on producing right stride when allocation domain has
   // transformation that cannot be represented via permutation. This is
   // currently used by PreprocessGroupedMatmulInputSf, where output is padded.
   std::set<IterDomain*> frontier_set(frontier.begin(), frontier.end());
   std::set<IterDomain*> logical_set(logical.begin(), logical.end());
   if (frontier_set != logical_set) {
-    return tensor;
+    std::vector<int64_t> logical_sizes(logical.size(), 0);
```

Comment:
IIUC, `logical_sizes` is the correct shape of the output, but `logical_strides` isn't. Can you leave a comment a little more about the context, e.g., why it should be done this way and why the incorrect strides don't matter.